packages/contracts-bedrock/scripts/DeployPeriphery.s.sol

```diff
@@ -0,0 +1,115 @@
+/// @notice Script used to deploy periphery contracts.
+contract DeployPeriphery is Deployer {
+    DeployPeripheryConfig cfg;
+
+    /// @notice The create2 salt used for deploying the faucet contract implementations.
+    ///         Using this helps to reduce config across networks as the implementation
+    ///         addresses will be the same across networks when deployed with create2.
+    bytes32 constant FAUCET_IMPL_SALT = keccak256(bytes("faucet contract salt"));
+
+    /// @notice The name of the script, used to ensure the right deploy artifacts
+    ///         are used.
+    function name() public pure override returns (string memory) {
+        return "DeployPeriphery";
+    }
+
+    function setUp() public override {
+        super.setUp();
+
+        string memory path = string.concat(vm.projectRoot(), "/deploy-config-periphery/", deploymentContext, ".json");
+        cfg = new DeployPeripheryConfig(path);
+
+        console.log("Deploying from %s", deployScript);
+        console.log("Deployment context: %s", deploymentContext);
+    }
+
+    /// @notice Deploy all of the periphery contracts
+    function run() public {
+        console.log("Deploying all periphery contracts");
+
+        deployProxies();
+        deployImplementations();
+
+        initializeFaucet();
+    }
+
+    /// @notice Deploy all of the proxies
+    function deployProxies() public {
+        deployProxyAdmin();
+
+        deployFaucetProxy();
+    }
+
+    /// @notice Deploy all of the implementations
+    function deployImplementations() public {
+        deployFaucet();
+    }
+
+    /// @notice Modifier that wraps a function in broadcasting.
+    modifier broadcast() {
+        vm.startBroadcast();
+        _;
+        vm.stopBroadcast();
+    }
+
+    /// @notice Deploy the ProxyAdmin
+    function deployProxyAdmin() public broadcast returns (address addr_) {
+        ProxyAdmin admin = new ProxyAdmin{ salt: keccak256(bytes("proxy admin salt")) }({
+            _owner: msg.sender
+        });
+        require(admin.owner() == msg.sender);
+
+        save("ProxyAdmin", address(admin));
+        console.log("ProxyAdmin deployed at %s", address(admin));
+        addr_ = address(admin);
+    }
+
+    /// @notice Deploy the FaucetProxy
+    function deployFaucetProxy() public broadcast returns (address addr_) {
+        address proxyAdmin = mustGetAddress("ProxyAdmin");
+        Proxy proxy = new Proxy{ salt: FAUCET_IMPL_SALT }({
+            _admin: proxyAdmin
+        });
+
+        address admin = address(uint160(uint256(vm.load(address(proxy), OWNER_KEY))));
+        require(admin == proxyAdmin);
+
+        save("FaucetProxy", address(proxy));
+        console.log("FaucetProxy deployed at %s", address(proxy));
+
+        addr_ = address(proxy);
+    }
+
+    /// @notice Deploy the faucet contract.
+    function deployFaucet() public broadcast returns (address) {
+        Faucet faucet = new Faucet{ salt: FAUCET_IMPL_SALT }(cfg.faucetAdmin());
+
```

Comment:
Is there a getter for the faucet admin? Would be nice to have a require for it here

Reply:
done